app/models/blog.rb

```diff
@@ -9,13 +9,21 @@ class Blog < ApplicationRecord
 
   scope :published, -> { where('secret = FALSE') }
 
-  scope :search, lambda { |term|
-    where("title LIKE '%#{term}%' OR content LIKE '%#{term}%'")
-  }
+  scope :viewable_by, ->(user) { published.or(where(user:)) }
+
+  scope :search, ->(term) { where('title LIKE :term OR content LIKE :term', term: "%#{term.present? ? sanitize_sql_like(term) : ''}%") }
```

Comment:
条件演算子による分岐部分がちょっとだけ読みづらいので、改善の候補を以下にいくつか書いておきます。2 個目の候補では `NilClass#to_s` が空文字を返すことを活用しています。

```suggestion
  scope :search, ->(term) { where('title LIKE :term OR content LIKE :term', term: "%#{sanitize_sql_like(term.presence || ''}%") }
```

```suggestion
  scope :search, ->(term) { where('title LIKE :term OR content LIKE :term', term: "%#{sanitize_sql_like(term.to_s}%") }
```

Reply:
候補のご提示ありがとうございます！🙏✨
`presence`も`nil.to_s`も把握していませんでした。どちらもとてもスマートですね😄
`presence`もまさにこの条件分岐のためのメソッドという感じで良いなと思ったのですが、`to_s`だとより簡潔であり、かつ文字列を期待する文脈であることがはっきりするからより良さそうだと思ったので、2個目の候補で修正させていただきました！

【参考】
- [#Object](https://api.rubyonrails.org/classes/Object.html#method-i-presence)
- [Rails: present?より便利なActiveSupportのpresenceメソッド（翻訳）｜TechRacho by BPS株式会社](https://techracho.bpsinc.jp/hachi8833/2021_02_26/62326)
- [NilClass\#to\_s \(Ruby 3\.3 リファレンスマニュアル\)](https://docs.ruby-lang.org/ja/latest/method/NilClass/i/to_s.html)